worlds/pokemon_emerald/test/TestAccessibility.py

```diff
@@ -0,0 +1,185 @@
+from Options import Toggle
+
+from . import PokemonEmeraldTestBase
+from ..util import location_name_to_label
+from ..options import NormanRequirement
+
+
+class TestBasic(PokemonEmeraldTestBase):
+    def testAlwaysAccessible(self) -> None:
+        self.assertTrue(self.can_reach_location(location_name_to_label("ITEM_ROUTE_102_POTION")))
+        self.assertTrue(self.can_reach_location(location_name_to_label("ITEM_ROUTE_115_SUPER_POTION")))
+
+
+class TestSurf(PokemonEmeraldTestBase):
+    options = {
+        "npc_gifts": Toggle.option_true
+    }
+
+    def testInaccessibleWithNoSurf(self) -> None:
+        self.assertFalse(self.can_reach_location(location_name_to_label("ITEM_PETALBURG_CITY_ETHER")))
+        self.assertFalse(self.can_reach_location(location_name_to_label("NPC_GIFT_RECEIVED_SOOTHE_BELL")))
+        self.assertFalse(self.can_reach_location(location_name_to_label("ITEM_LILYCOVE_CITY_MAX_REPEL")))
+        self.assertFalse(self.can_reach_entrance("REGION_ROUTE118/WATER -> REGION_ROUTE118/EAST"))
+        self.assertFalse(self.can_reach_entrance("REGION_ROUTE119/UPPER -> REGION_FORTREE_CITY/MAIN"))
+        self.assertFalse(self.can_reach_entrance("MAP_FORTREE_CITY:3/MAP_FORTREE_CITY_MART:0"))
+
+    def testAccessibleWithSurfOnly(self) -> None:
+        self.collect_by_name(["HM03 Surf", "Balance Badge"])
+        self.multiworld.state.sweep_for_events()
```

Comment:
CollectionState should do a sweep when you call `collect_by_name` already

Reply:
Fixed in cb981ec